lang/java/compiler/src/main/javacc/org/apache/avro/compiler/idl/idl.jj

```diff
@@ -1020,13 +1020,17 @@ TOKEN :
 Protocol CompilationUnit():
 {
   Protocol p;
+  DocCommentHelper.getAndClearWarnings(); // Throw away previous results.
 }
 {
   p = ProtocolDeclaration()
   ( < "\u001a" > )?
   ( <STUFF_TO_IGNORE: ~[]> )?
   <EOF>
-  { return SchemaResolver.resolve(p); }
+  {
+    parserWarnings = DocCommentHelper.getAndClearWarnings();
```

Comment:
Lines 1023 and 1031 are executed in the same method, and hence always on the same thread. This guarantees that between these lines, the use of `ThreadLocal` fields in the class `DocCommentHelper` works the same way as instance fields.

(note: we need to use `ThreadLocal` fields, because the generated class `IdlTokenManager` needs to store documentation somewhere (due to the leniency of the parser), and we cannot we cannot add instance fields to it)